### PR TITLE
Remove references to schema1 pacakge from storage

### PR DESCRIPTION
**Pull request by DavidSpek**
Part of: https://github.com/distribution/distribution/issues/3966

This PR is a step towards the elimination of `schema v1` support in `distribution`.

This PR removes schema1 package references from the `registry/storage` package.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (milosgajdos) [inline renovate.json:1]: this file must go to a different PR

**Review by milosgajdos — CHANGES_REQUESTED** [review]

File: renovate.json (hunk 1 above)
Comment (DavidSpek) [reply renovate.json]: Was a mistake with rebasing, it's gone now.

File: registry/storage/garbagecollect_test.go [hunk 2]
```diff
@@ -23,11 +22,7 @@ type image struct {
 
 func createRegistry(t *testing.T, driver driver.StorageDriver, options ...RegistryOption) distribution.Namespace {
 	ctx := context.Background()
-	k, err := libtrust.GenerateECP256PrivateKey()
-	if err != nil {
-		t.Fatal(err)
-	}
-	options = append([]RegistryOption{EnableDelete, Schema1SigningKey(k), EnableSchema1}, options...)
+	options = append([]RegistryOption{EnableDelete}, options...)
```
Comment (milosgajdos) [inline registry/storage/garbagecollect_test.go:25]: Let's make this more idiomatic `Go`:

```suggestion
	options = append(options, EnableDelete)
```

**Review by milosgajdos — APPROVED** [review]
I have a small suggestion, otherwise LGTM.

File: registry/storage/manifeststore_test.go [hunk 3]
```diff
@@ -79,13 +66,59 @@ func testManifestStorage(t *testing.T, schema1Enabled bool, options ...RegistryO
...
+				Size:      6863,
+				MediaType: schema2.MediaTypeLayer,
+			},
 		},
-		Name: env.name.Name(),
-		Tag:  env.tag,
 	}
+	sampleConfigDigest := digest.FromBytes(sampleConfig)
+	sampleConfigSize := int64(len(sampleConfig))
+	m.Config.Digest = sampleConfigDigest
```
Comment (Jamstah) [inline registry/storage/manifeststore_test.go:120]: Should we put these straight into the manifest?

File: registry/storage/garbagecollect_test.go [hunk 4]
```diff
@@ -23,11 +22,7 @@ type image struct {
 
 func createRegistry(t *testing.T, driver driver.StorageDriver, options ...RegistryOption) distribution.Namespace {
 	ctx := context.Background()
-	k, err := libtrust.GenerateECP256PrivateKey()
-	if err != nil {
-		t.Fatal(err)
-	}
-	options = append([]RegistryOption{EnableDelete, Schema1SigningKey(k), EnableSchema1}, options...)
+	options = append(options, EnableDelete)
```
[the remark this answers is not in the document]
Comment (milosgajdos) [reply registry/storage/garbagecollect_test.go:25]: It's only scoped to each individual test that needs to create a registry object - these options are not used anywhere else so it's ok, imho, but yeah we can change it back.

File: registry/storage/garbagecollect_test.go [hunk 5]
```diff
@@ -215,11 +186,21 @@ func TestDeleteManifestIfTagNotFound(t *testing.T) {
 		t.Fatalf("failed to make layers: %v", err)
 	}
 
+	digests1 := []digest.Digest{}
```
[the remark this answers is not in the document]
Comment (milosgajdos) [reply registry/storage/garbagecollect_test.go:189]: the original coode was calling `getKeys(randomLayers1)` and `getKeys(randomLayers2)` when passing the params to `estutil.MakeSchema1Manifest`; we could have kept that helper maybe; this code just makes it explicit because `getKeys` was removed (I'm assuming as part of the removal of `testutil.MakeSchema1Manifest` calls, and then never put back leading to this code. But this code is needed to pass the digests down to `testutil.MakeSchema2Manifest` instead of the schema1 `estutil.MakeSchema1Manifest` call. Maybe a bit of an oversight but not incorrect. @DavidSpek do you wanna put back `getKeys` and then pass `getKeys(randomlayers[12])` down to schema2 manifet func

File: registry/storage/garbagecollect_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (Jamstah) [reply registry/storage/garbagecollect_test.go:25]: I agree in test code it's really a very minor nit :)

File: registry/storage/garbagecollect_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jamstah) [reply registry/storage/garbagecollect_test.go:189]: That comment was more for my understanding than a request for change, happy to go with either style.

[the remark this answers is not in the document]
Comment (milosgajdos) [reply registry/storage/garbagecollect_test.go:189]: I think we should put that func (getKeys) back either way. This is a great catch even if we arrived at it by tangent 🙃

File: registry/storage/manifeststore_test.go [hunk 6]
```diff
@@ -79,13 +66,59 @@ func testManifestStorage(t *testing.T, schema1Enabled bool, options ...RegistryO
...
+			MediaType:     schema2.MediaTypeManifest,
+		},
+		Config: distribution.Descriptor{
+			Digest:    "sha256:1a9ec845ee94c202b2d5da74a24f0ed2058318bfa9879fa541efaecba272e86b",
+			Size:      3253,
+			MediaType: schema2.MediaTypeImageConfig,
+		},
+		Layers: []distribution.Descriptor{
+			{
+				Digest:    "sha256:463434349086340864309863409683460843608348608934092322395278926a",
```
[the remark this answers is not in the document]
Comment (DavidSpek) [reply registry/storage/manifeststore_test.go:107]: These are just placeholder layers that get filled in by the loop that generates random tar files later.

File: registry/storage/manifeststore_test.go (hunk 3 above)
Comment (DavidSpek) [reply registry/storage/manifeststore_test.go:120]: That might clean things up a bit yes.

File: registry/storage/manifeststore_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (Jamstah) [reply registry/storage/manifeststore_test.go:107]: Can we have a comment saying they are placeholders, or make them "sha256:000000...", or preferably just build the object after we know the values?

File: registry/storage/manifeststore_test.go (hunk 3 above)
Comment (DavidSpek) [reply registry/storage/manifeststore_test.go]: Cleaned it up :)

File: registry/storage/garbagecollect_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (DavidSpek) [reply registry/storage/garbagecollect_test.go]: Added back the `getKeys()` function and am passing that to `testutil.MakeSchema2Manifest` now.

**Review by Jamstah — APPROVED** [review]
Much cleaner, thanks.
